include/oneapi/mkl/dft/descriptor.hpp

```diff
@@ -76,6 +89,98 @@ class descriptor {
     oneapi::mkl::dft::dft_values values;
 };
 
+template <precision prec, domain dom>
+descriptor<prec, dom>::descriptor(std::vector<std::int64_t> dimension) :
+    dimension_(dimension),
+    handle_(nullptr),
+    rank_(dimension.size())
+    {
+        // TODO: initialize the device_handle, handle_buffer
+        values.domain = dom;
+        values.precision = prec;
+        values.dimension = dimension_;
+        values.rank = rank_;
+    }
```

Comment:
I have:
```
        // Compute default strides.
        std::vector<std::int64_t> defaultStrides(rank_, 1);
        for(int i{rank - 1}; i < 0; --i){
            defaultStrides[i] = defaultStrides[i - 1] * dimension_[i];
        }
        defaultStrides[0] = 0;
        values_.input_strides = defaultStrides;
        values_.output_strides = defaultStrides;
        values_.bwd_scale = 1.0;
        values_.fwd_scale = 1.0;
        values_.number_of_transform = 1;
        values_.fwd_dist = 0;
        values_.bwd_dist = 0;
        values_.placement = config_value::INPLACE;
        values_.complex_storage = config_value::COMPLEX_COMPLEX;
        values_.conj_even_storage = config_value::COMPLEX_COMPLEX; 
        values_.dimension = rank_;
        values_.domain = dom;
        values_.precision = prec;
```
although I haven't been able to test this yet.

Reply:
`fwd,bwd` distances cannot be 0, but I have integerated the rest of this into the constructor on my end